main.go

```diff
@@ -393,9 +393,12 @@ func (c *config) addTags(fieldName string, tags *structtag.Tags) (*structtag.Tag
 
 	for _, key := range c.add {
 		splitted = strings.Split(key, ":")
-		if len(splitted) == 2 {
+		if len(splitted) >= 2 {
 			key = splitted[0]
 			name = splitted[1]
+			for i := 2; i < len(splitted); i++ {
+				name += ":" + splitted[i]
+			}
```

Comment:
This will fail for these use cases:

```
json:MyBar=bar
json:MyBar=bar=foo
json:MyBar:bar:foo=qux 
```

Those are all valid. After splitting, the first item should be the `key` and the rest `name`. The below code works for all these cases, including your test case

```go
splitted = strings.SplitN(key, ":", 2)
if len(splitted) >= 2 {
	key = splitted[0]
	name = strings.Join(splitted[1:], "")
} ...
```

Can you add the above test cases and updated your PR?

Reply:
Hi,
added those test cases and made the refactoring.
updated the pr